test/SkipCallsTestHook.t.sol

```diff
@@ -0,0 +1,255 @@
+
+import {Test} from "forge-std/Test.sol";
+import {Vm} from "forge-std/Vm.sol";
+import {PoolId, PoolIdLibrary} from "../src/types/PoolId.sol";
+import {Hooks} from "../src/libraries/Hooks.sol";
+import {SwapFeeLibrary} from "../src/libraries/SwapFeeLibrary.sol";
+import {IPoolManager} from "../src/interfaces/IPoolManager.sol";
+import {IProtocolFees} from "../src/interfaces/IProtocolFees.sol";
+import {IHooks} from "../src/interfaces/IHooks.sol";
+import {PoolKey} from "../src/types/PoolKey.sol";
+import {PoolManager} from "../src/PoolManager.sol";
+import {PoolSwapTest} from "../src/test/PoolSwapTest.sol";
+import {Deployers} from "./utils/Deployers.sol";
+import {GasSnapshot} from "forge-gas-snapshot/GasSnapshot.sol";
+import {Currency, CurrencyLibrary} from "../src/types/Currency.sol";
+import {MockERC20} from "solmate/test/utils/mocks/MockERC20.sol";
+import {Constants} from "../test/utils/Constants.sol";
+import {SkipCallsTestHook} from "../src/test/SkipCallsTestHook.sol";
+
+contract SkipCallsTest is Test, Deployers, GasSnapshot {
+    using PoolIdLibrary for PoolKey;
+
+    IPoolManager.SwapParams swapParams =
+        IPoolManager.SwapParams({zeroForOne: true, amountSpecified: -100, sqrtPriceLimitX96: SQRT_RATIO_1_2});
+
+    PoolSwapTest.TestSettings testSettings =
+        PoolSwapTest.TestSettings({withdrawTokens: true, settleUsingTransfer: true, currencyAlreadySent: false});
+
+    function deploy(SkipCallsTestHook skipCallsTestHook) private {
+        SkipCallsTestHook impl = new SkipCallsTestHook();
+        vm.etch(address(skipCallsTestHook), address(impl).code);
+        deployFreshManagerAndRouters();
+        skipCallsTestHook.setManager(IPoolManager(manager));
+
+        (currency0, currency1) = deployMintAndApprove2Currencies();
+
+        assertEq(skipCallsTestHook.counter(), 0);
+
+        (key,) = initPool(currency0, currency1, IHooks(address(skipCallsTestHook)), 3000, SQRT_RATIO_1_1, ZERO_BYTES);
+    }
+
+    function approveAndAddLiquidity(SkipCallsTestHook skipCallsTestHook) private {
+        MockERC20(Currency.unwrap(key.currency0)).approve(address(skipCallsTestHook), Constants.MAX_UINT256);
+        MockERC20(Currency.unwrap(key.currency1)).approve(address(skipCallsTestHook), Constants.MAX_UINT256);
+        modifyLiquidityRouter.modifyLiquidity(key, LIQ_PARAMS, abi.encode(address(this)));
+    }
+
+    function test_beforeInitialize_skipIfCalledByHook() public {
+        SkipCallsTestHook skipCallsTestHook = SkipCallsTestHook(
+            address(
+                uint160(0xFFfFfFffFFfffFFfFFfFFFFFffFFFffffFfFFFfF)
+                    & uint160(
+                        ~Hooks.AFTER_INITIALIZE_FLAG & ~Hooks.BEFORE_ADD_LIQUIDITY_FLAG & ~Hooks.AFTER_ADD_LIQUIDITY_FLAG
+                            & ~Hooks.BEFORE_REMOVE_LIQUIDITY_FLAG & ~Hooks.AFTER_REMOVE_LIQUIDITY_FLAG & ~Hooks.BEFORE_SWAP_FLAG
+                            & ~Hooks.AFTER_SWAP_FLAG & ~Hooks.BEFORE_DONATE_FLAG & ~Hooks.AFTER_DONATE_FLAG
```

Comment:
so these lines can become
```
        address(
            uint160(
                uint256(type(uint160).max) & clearAllHookPermisssionsMask | Hooks.AFTER_INITIALIZE_FLAG
            )
        )
```

Reply:
not sure why it needs to go via uint256... you could try making `clearAllHookPermisssionsMask` a uint160 too

Reply:
i think you commented something similar in #548 which I can do over there

Reply:
ah so much nicer! thank you!!